### DISCUSSION TITLE
Automatically establishing server-side datachannel

### DISCUSSION THREAD

**Discussion by unicomp21** [pion/webrtc#2907] [Ideas]
When using nothing but datachannel, given an offer, is it possible to predict the answer?  ie if answer could be predicted in browser/js, then one could establish lossy udp/datachannel w/ backend pretty easy/fast? @Sean-Der

**Comment by Sean-Der** [comment]
Yea! Have you seen https://github.com/pion/offline-browser-communication @unicomp21 ?

You can do it Go<->Go, not supported in browsers though.

**Reply by unicomp21** [reply]
What if the offer was passed via uname/pwd in stun logon via some length delimited encoding/format (text based)?

**Reply by unicomp21** [reply]
I need it in browser, is it possible to do something like this:

https://github.com/paullouisageneau/violet/issues/22
